doc/nrf/ug_matter_device_security.rst

```diff
@@ -0,0 +1,17 @@
+.. _ug_matter_device_security:
+
+Securing devices
+################
+
+.. contents::
+   :local:
+   :depth: 2
+
+.. include:: app_approtect.rst
+   :start-after: app_approtect_info_start
+   :end-before: app_approtect_info_end
+
+See :ref:`app_approtect` for more information.
+
+If you are working on a Matter over Thread end product based on the nRF52840 or the nRF5340 SoC, make sure to disable this mechanism to debug the firmware.
```

Comment:
It's a little ambiguous if we do suggest to enable or disable this feature. Maybe something like this (i believe user will figure out to disable AP-Protect (or not enabling) to have debug functionality. Btw. it's not something specific to Matter over Thread/Wi-Fi but to SoC in general.

```suggestion
Make sure to enable AP-Protect feature for the devices in the production to disable debug functionality. 
```